### PR TITLE
Linux: Add and install app metadata file

### PR DESCRIPTION
**Pull request by guihkx**
This metadata file dictates how your app will be presented to users on Linux app stores.

[Here's an example](https://flathub.org/apps/im.bernard.Memorado) of a random app on Flathub, and [here's the app metainfo file](https://github.com/wbernard/Memorado/blob/10536b1d62f977b4452e9c20f50a160dc46cf829/data/im.bernard.Memorado.metainfo.xml.in.in) for reference.

Documentation: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html

---

This is (hopefully) the last missing piece for https://github.com/horsicq/Detect-It-Easy/issues/189!

### CODE REVIEW

File: LINUX/io.github.horsicq.detect-it-easy.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    <p>It detects, among other things, the compiler and linker used, signatures, and other information about files.</p>
+    <p>The program has been designed specifically to reveal file signatures and how they are packed.</p>
+    <p>Some of the binary formats supported:</p>
+    <ul>
+      <li>MS-DOS: DOS MZ</li>
+      <li>Windows: Portable Executable (PE)</li>
+      <li>*Nix: Executable and Linkable Format (ELF)</li>
+      <li>macOS: Mach Object (Mach-O)</li>
+    </ul>
+  </description>
```
Comment (guihkx) [inline LINUX/io.github.horsicq.detect-it-easy.metainfo.xml:22]: Let me know if this description of your app does it justice.

File: LINUX/io.github.horsicq.detect-it-easy.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  <launchable type="desktop-id">io.github.horsicq.detect-it-easy.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/guihkx/flatpak-app-detect-it-easy/ef62b241881138f43ce7004b5a6760d5f6c24deb/screenshots/main_window_basic_view.png</image>
```
Comment (guihkx) [inline LINUX/io.github.horsicq.detect-it-easy.metainfo.xml:26]: This screenshot, and the one below this one, are currently being hosted [on my repository](https://github.com/guihkx/flatpak-app-detect-it-easy/tree/master/screenshots), which is not ideal, but should be fine for now.

In any case, let me know if you have a better place to host them right now (or if you just want take different screenshots yourself).

Here's the [recommended guidelines](https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html#qsr-app-screenshots-info) for them.

File: LINUX/io.github.horsicq.detect-it-easy.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      <caption>Disassembly window (advanced view)</caption>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://horsicq.github.io/#detect-it-easydie</url>
+  <url type="bugtracker">https://github.com/horsicq/Detect-It-Easy/issues</url>
+  <content_rating type="oars-1.1"/>
+  <releases>
+    <release version="3.10" date="2024-11-03">
+      <url>https://raw.githubusercontent.com/horsicq/Detect-It-Easy/refs/heads/master/changelog.txt</url>
+    </release>
```
Comment (guihkx) [inline LINUX/io.github.horsicq.detect-it-easy.metainfo.xml:40]: This is arguably the most important tag here.

Before releasing a new version, you must **always** update this section.

You can achieve that in two ways:

1\. Add a new `<release>` tag at the top, for example:

```xml
<releases>
  <release version="3.11" date="2025-01-03">
    <url>https://raw.githubusercontent.com/horsicq/Detect-It-Easy/refs/heads/master/changelog.txt</url>
  </release>
  <release version="3.10" date="2024-11-03">
    <url>https://raw.githubusercontent.com/horsicq/Detect-It-Easy/refs/heads/master/changelog.txt</url>
  </release>
</releases>
```

2\. Keep just a single `<release>` tag, for example:

```xml
<releases>
  <release version="3.11" date="2025-01-03">
    <url>https://raw.githubusercontent.com/horsicq/Detect-It-Easy/refs/heads/master/changelog.txt</url>
  </release>
</releases>
```

File: build_dpkg.sh [hunk 4]
```diff
@@ -47,6 +47,7 @@ if [ -z "$X_ERROR" ]; then
         cp -f $X_SOURCE_PATH/build/release/diel                             $X_SOURCE_PATH/release/$X_BUILD_NAME/usr/bin/
         cp -f $X_SOURCE_PATH/LINUX/io.github.horsicq.detect-it-easy.desktop $X_SOURCE_PATH/release/$X_BUILD_NAME/usr/share/applications/
         cp -Rf $X_SOURCE_PATH/LINUX/hicolor/                                $X_SOURCE_PATH/release/$X_BUILD_NAME/usr/share/icons/
+        cp -f $X_SOURCE_PATH/LINUX/io.github.horsicq.detect-it-easy.metainfo.xml $X_SOURCE_PATH/release/$X_BUILD_NAME/usr/share/metainfo/
```
Comment (guihkx) [inline build_dpkg.sh:50]: This change messes up the indentation of this section...

If you think that doesn't look good, let me know and I'll add extra whitespaces to the lines above to align the two columns.

File: create_appimage.sh [hunk 5]
```diff
@@ -12,6 +12,7 @@ cp -f $X_SOURCE_PATH/build/release/die                              $X_SOURCE_PA
 #cp -f $X_SOURCE_PATH/build/release/diel                             $X_SOURCE_PATH/release/appDir/usr/bin/
 cp -f $X_SOURCE_PATH/LINUX/io.github.horsicq.detect-it-easy.desktop $X_SOURCE_PATH/release/appDir/usr/share/applications/
 cp -Rf $X_SOURCE_PATH/LINUX/hicolor/                                $X_SOURCE_PATH/release/appDir/usr/share/icons/
+cp -f $X_SOURCE_PATH/LINUX/io.github.horsicq.detect-it-easy.metainfo.xml $X_SOURCE_PATH/release/appDir/usr/share/metainfo/
```
Comment (guihkx) [inline create_appimage.sh:15]: This change messes up the indentation of this section...

If you think that doesn't look good, let me know and I'll add extra whitespaces to the lines above to align the two columns.

File: install.sh [hunk 6]
```diff
@@ -18,6 +19,7 @@ cp -f $X_SOURCE_PATH/build/release/diec                             $X_PREFIX/bi
 cp -f $X_SOURCE_PATH/build/release/diel                             $X_PREFIX/bin/
 cp -f $X_SOURCE_PATH/LINUX/io.github.horsicq.detect-it-easy.desktop $X_PREFIX/share/applications/
 cp -Rf $X_SOURCE_PATH/LINUX/hicolor/                                $X_PREFIX/share/icons/
+cp -f $X_SOURCE_PATH/LINUX/io.github.horsicq.detect-it-easy.metainfo.xml $X_PREFIX/share/metainfo/
```
Comment (guihkx) [inline install.sh:22]: This change messes up the indentation of this section...

If you think that doesn't look good, let me know and I'll add extra whitespaces to the lines above to align the two columns.

File: LINUX/io.github.horsicq.detect-it-easy.metainfo.xml (hunk 1 above)
Comment (horsicq) [reply LINUX/io.github.horsicq.detect-it-easy.metainfo.xml:22]: The description is awesome. Thanks! :)

File: LINUX/io.github.horsicq.detect-it-easy.metainfo.xml (hunk 2 above)
Comment (horsicq) [reply LINUX/io.github.horsicq.detect-it-easy.metainfo.xml:26]: It is Ok! :)

File: LINUX/io.github.horsicq.detect-it-easy.metainfo.xml (hunk 3 above)
Comment (horsicq) [reply LINUX/io.github.horsicq.detect-it-easy.metainfo.xml:40]: ok!

File: build_dpkg.sh (hunk 4 above)
Comment (horsicq) [reply build_dpkg.sh:50]: It is good!

**Comment by guihkx** [comment]
Thanks for merging!

Please note that these changes also depend on https://github.com/horsicq/build_tools/commit/895f03f860dd7e4f661ad7a6e2a9cda8a5f4e3ce.

But I notice you update all submodules frequently so that should be taken care of soon. :)

**Pull request merged**